### ISSUE TITLE
`Functor (m k) => Map m k` superclass

### ISSUE THREAD

**Issue by tysonzero** [arybczak/list-tries#11]
Similar to how `Foldable (m k)` is a superclass, it seems like `Functor (m k)` would also be a useful superclass instead of requiring a custom `map` function.

**Comment by Deewiant** [comment]
Yeah, it'd make sense. Though the need for `mapWithKey` means that it won't make much of a difference in practice though. Similarly `Traversable (m k)` would allow getting rid of `mapAccum` but not `mapAccumWithKey`.
